packages/utils-query-operation/lib/FragmentSelectorShapes.ts

```diff
@@ -76,11 +82,18 @@ function doesShapeAcceptOperationRecurseOperation(
       return false;
     }
   }
-  if (operation.patterns && !operation.patterns
-    .every((input: Algebra.Pattern) => doesShapeAcceptOperationRecurseShape(shapeTop, shapeTop, input, options))) {
-    return false;
-  }
-  return true;
+  return !(operation.patterns && !operation.patterns
+    .every((input: Algebra.Pattern) => doesShapeAcceptOperationRecurseShape(shapeTop, shapeTop, input, options)));
+}
+
+function isBuiltInFunctionIRI(iri: string): boolean {
+  return iri.includes('w3.org/2005/xpath-functions#') ||
```

Comment:
@jitsedesmet Is this a safe check?

Reply:
I will need to look in greater detail, but I don't think so. These named functions are supported by Comunica. [The spec](https://www.w3.org/TR/sparql11-query/#FunctionMapping) asks for these constructor functions to be implemented. But Many named functions can be suppported. I have not looked but maybe the better approach is just to ask the bus in Comunica for a function evaluator. - If the bus accepts, we have a function, otherwise not. Anyways, I will look closer later.